apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/ConfigsExportService.java

```diff
@@ -0,0 +1,187 @@
+package com.ctrip.framework.apollo.portal.service;
+
+import com.ctrip.framework.apollo.common.dto.ClusterDTO;
+import com.ctrip.framework.apollo.common.entity.App;
+import com.ctrip.framework.apollo.core.enums.ConfigFileFormat;
+import com.ctrip.framework.apollo.portal.component.PermissionValidator;
+import com.ctrip.framework.apollo.portal.component.PortalSettings;
+import com.ctrip.framework.apollo.portal.entity.bo.ConfigBO;
+import com.ctrip.framework.apollo.portal.entity.bo.NamespaceBO;
+import com.ctrip.framework.apollo.portal.environment.Env;
+import com.ctrip.framework.apollo.portal.util.ConfigFileUtils;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ConfigsExportService {
+
+  private final static Logger logger = LoggerFactory.getLogger(ConfigsExportService.class);
+
+  private final AppService appService;
+
+  private final ClusterService clusterService;
+
+  private final NamespaceService namespaceService;
+
+  private final PortalSettings portalSettings;
+
+  private final PermissionValidator permissionValidator;
+
+  public ConfigsExportService(
+      AppService appService,
+      ClusterService clusterService,
+      final @Lazy NamespaceService namespaceService,
+      PortalSettings portalSettings,
+      PermissionValidator permissionValidator) {
+    this.appService = appService;
+    this.clusterService = clusterService;
+    this.namespaceService = namespaceService;
+    this.portalSettings = portalSettings;
+    this.permissionValidator = permissionValidator;
+  }
+
+  /**
+   * write multiple namespace to a zip.
+   * use {@link Stream#reduce(Object, BiFunction, BinaryOperator)} to forbid concurrent write.
+   * @param configBOStream namespace's stream
+   * @param outputStream receive zip file output stream
+   * @throws IOException
+   */
+  private static void writeAsZipOutputStream(Stream<ConfigBO> configBOStream, OutputStream outputStream)
```

Comment:
Is it necessarily marked as `static`?

Reply:
The method write `Stream<ConfigBO>`'s content to an `OutputStream`. 

Everytime the method is invoked, those parameters are different with before, so I think use `static` for this method is better.